src/main/kotlin/com/avail/descriptor/representation/AbstractDescriptor.kt

```diff
@@ -3808,6 +3819,9 @@ abstract class AbstractDescriptor protected constructor (
 
 	abstract fun o_RegisterDump (self: AvailObject): AvailObject
 
+	abstract fun o_ModuleAddStyler(self: AvailObject, styler: A_Styler)
```

Comment:
The "real" version is the extension method `A_Module.moduleAddStyler()` in `A_Module`.  You can get there by Cmd-clicking `o_ModuleAddStyler`.  You can get to the meat of the implementation by clicking in o_ModuleAddStyler and typing Cmd-Option-B, then choosing the one that *isn't* in Descriptor or IndirectionDescriptor.

I have intentionally *not* been including comments for new methods in `AbstractDescriptor`, `Descriptor`, and `IndirectionDescriptor`.  We intend to simply generate the majority of these files when we have the right technology (*ahem*, Rust macros).  Instead, the comments go into the `A_*` methods/properties.  These are the methods/properties that are directly referenced from primitives, the interpreter, and other subsystems that use AvailObjects.

Besides the comments in `A_*` methods/properties, we also document the slot (field) definitions in `...Descriptor.ObjectSlots` and `...Descriptor.IntegerSlots`.  These describe the layout of objects that use the enclosing descriptor class.  We use reflection and annotations in stylized ways to reduce the boilerplate.

All this noisy code exists so that:
1. We can have multiple representations for things like tuples, allowing efficient subrange extraction and concatenation, as well as compact strings.
2. We can change an object's representation and behavior dynamically by altering its descriptor, such as changing it to an IndirectionDescriptor, or switching from a mutable descriptor instance to an immutable or shared instance of the same descriptor class.

Reply:
Ok, makes sense. Especially if we decide to make the jump to Rust, of course, given enough momentum from releasing Anvil which will undoubtedly make a port much easier.